### ISSUE TITLE
Sometimes autocomplete results in duplication of the first few letters of the function name

### ISSUE THREAD

**Issue by ronblum** [rstudio/rstudio#5979]
### System details

    RStudio Edition : Desktop (and Server , I think)
    RStudio Version : 1.x
    OS Version      : Multiple
    R Version       : Multiple

### Steps to reproduce the problem

The issue is intermittent—I can't figure out a consistently reproducible example. But when it does happen:

1. In the console, type the beginning of a command. For example, `lengt`.
2. Hit <kbd>tab</kbd> for autocomplete.
3. Choose `length`.

### Describe the problem in detail

Sometimes, intermittently, the result is something like `lelength()`.

### Describe the behavior you expected

The result should be just the name of the function—in this example, `length()`.

- [x ] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x ] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [ ] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x ] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kevinushey** [comment]
Here's a way to more reliably reproduce the issue. (I think it's easier to reproduce when there is a bit of latency; e.g. when using RStudio Server.)

1. Type `rn` in the source editor.
2. Press <kbd>Tab</kbd> to request completions.
3. Quickly type `om`, before the completion list has a chance to display.
4. Accept the completion that is presented on the screen before it is dismissed.

In this case, I end up with the completion `rnrnbinom()` in the editor, rather than just `rbinom()`.

**Comment by calpan** [comment]
This happens frequently for me with filename autocompletion when there is some noticeable lag between hitting Tab and seeing the popup list (possibly due to large number of files in target folders).

**Comment by kevinushey** [comment]
Here's a way to reproduce this more consistently:

```
# simulate a 'laggy' autocompletion request
trace(.rs.rpc.get_completions, quote(Sys.sleep(1)), print = FALSE)
```

Then, type the following:

```
stats::ac<TAB>
```

Wait for the autocompletion list to pop up, and then accept the first completion result. You should see something like the following:

![acf](https://user-images.githubusercontent.com/1976582/94471716-78da1f80-017e-11eb-93c9-1882d649be04.gif)

**Comment by kevinushey** [comment]
Another case where this can happen:

![oops](https://user-images.githubusercontent.com/1976582/94472980-6a8d0300-0180-11eb-99a7-f3a849a6242d.gif)

**Comment by kevinushey** [comment]
Another case where this can happen:

![oops](https://user-images.githubusercontent.com/1976582/94473090-91e3d000-0180-11eb-9b4a-ea205cfa6e24.gif)

**Comment by kfeinauer** [comment]
Verified fixed 1.4.944
